packages/kbn-es-types/src/search.ts

```diff
@@ -602,7 +603,9 @@ type WrapAggregationResponse<T> = keyof UnionToIntersection<T> extends never
 
 export type InferSearchResponseOf<
   TDocument = unknown,
-  TSearchRequest extends estypes.SearchRequest = estypes.SearchRequest,
+  TSearchRequest extends
+    | estypes.SearchRequest
+    | (estypesWithoutBodyKey.SearchRequest & { body?: never }) = estypes.SearchRequest,
```

Comment:
@elastic/apm-ui with the `typesWithBodyKey` definition no longer having fields declared as top level properties, the whole inferred types kinda exploded. 

The only way I found to fix it is to have the `TSearchRequest` generic type extending either the `typesWithBodyKey` or the `types` definition and make the `hits` inference based on whether one of the other was used. 

It didn't cause any downstream problems, so I think it's fine, but just wanted to highlight that.

Reply:
If the types still work (as in, they are not downgraded to `any`, `unknown` or `never`) that is fine with me.